$OpenBSD: patch-source_common_putilimp_h,v 1.6 2013/03/24 10:28:35 ajacoutot Exp $

hppa and sparc don't have atomic builtins:
__sync_val_compare_and_swap, __sync_add_and_fetch, __sync_sub_and_fetch
don't assume having GNUC >= 401 implies they're here..

--- source/common/putilimp.h.orig	Fri Mar 15 23:11:58 2013
+++ source/common/putilimp.h	Sun Mar 24 11:07:42 2013
@@ -126,6 +126,8 @@ typedef size_t uintptr_t;
    /* not defined */
 #elif U_PLATFORM == U_PF_OS400
    /* not defined */
+#elif U_PLATFORM == U_BSD
+   /* not defined */
 #else
 #   define U_TIMEZONE timezone
 #endif
@@ -183,7 +185,7 @@ typedef size_t uintptr_t;
  */
 #ifdef U_HAVE_GCC_ATOMICS
     /* Use the predefined value. */
-#elif U_GCC_MAJOR_MINOR >= 404
+#elif U_GCC_MAJOR_MINOR >= 404 && !defined(__hppa__) && (defined(__sparc__) && ! defined(__sparcv9__))
 #   define U_HAVE_GCC_ATOMICS 1
 #else
 #   define U_HAVE_GCC_ATOMICS 0
